src/components/entity/state-info.ts

```diff
@@ -84,7 +84,7 @@ class StateInfo extends LitElement {
     }
 
     const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
-    if (!oldHass || oldHass.language !== this.hass.language) {
+    if (!oldHass || oldHass.locale.language !== this.hass.locale.language) {
```

Comment:
We should also rerender when another local option changes
```suggestion
    if (!oldHass || oldHass.locale !== this.hass.locale) {
```